### ISSUE TITLE
BUG: base_model does not accept local filesystem paths

### ISSUE THREAD

**Issue by vijaydevmasters** [f-inverse/jammi-ai#233]
### WHAT HAPPENED
Called `db.fine_tune()` with a local model directory as `base_model`. Two formats were tried - both failed with the same error:

**Attempt 1: bare absolute path**
```python
db.fine_tune(
    base_model="/home/.../models_nexus/ModernBERT",
    ...
)
```
**Error:**
```text
Model error: /home/.../models_nexus/ModernBERT:
Failed to download config: request error: http status: 404
```

**Attempt 2: file:// URI**
```python
db.fine_tune(
    base_model="file:///home/.../models_nexus/ModernBERT",
    ...
)
```
**Error:**
```text
Model error: file:///home/.../models_nexus/ModernBERT:
Failed to download config: request error: http status: 404
```

The model files existed locally and were valid (same files used successfully by PyTorch / transformers in a separate script). The 404 is not a missing file - it is Jammi constructing a HuggingFace Hub URL from the path string and hitting the HF API, which returns 404 because the path is not a valid Hub model ID.

---

### ROOT CAUSE (CONFIRMED by source inspection)
In `jammi_client/_assembly.py`, the `build_fine_tune_request()` function passes `base_model` verbatim into the protobuf with no processing whatsoever:

```python
return training_pb2.StartTrainingRequest(
    fine_tune=training_pb2.FineTuneSpec(
        source=source,
        columns=list(columns),
        method=wire_method,
        task=wire_task,
    ),
    base_model=base_model,   # <-- passed through unchanged
    config=config,
)
```

The Rust engine's `hf-hub` crate then receives whatever string was passed and always resolves it as a HuggingFace Hub model ID, constructing a URL of the form:
`https://huggingface.co/<base_model>/resolve/main/config.json`

* A bare path like `/home/user/model` becomes:
  `https://huggingface.co//home/user/model/resolve/main/config.json` -> **404**

* A `file://` URI like `file:///home/user/model` becomes:
  `https://huggingface.co/file:///home/user/model/resolve/main/config.json` -> **404**

There is no code path in either the Python layer or the Rust hf-hub resolver that detects a local path and reads model files directly from disk.

---

### CONTRAST WITH add_source()
The `add_source()` verb DOES have local path support. In `jammi_client/_assembly.py` there is an explicit helper:

```python
def _local_source_url(url: str) -> str:
    if "://" in url:
        return url
    return "file://" + url
```

This normalises bare paths into `file://` URLs and the Rust engine reads local files for sources. No equivalent exists for `base_model`.

---

### IMPACT
Users in air-gapped environments, private cloud VMs, or who have fine-tuned backbone models stored locally have no supported way to use them. They are forced to either:
1. Push their private model to the public HuggingFace Hub (not always acceptable for proprietary or regulated data environments), or
2. Manually construct a fake HF hub cache directory structure on disk, set `HF_HUB_CACHE` and `HF_HUB_OFFLINE=1` env vars before the native `.so` loads, and use the public HF model ID as a proxy key - an undocumented, fragile workaround that requires understanding Jammi's internal caching internals.

---

### EXPECTED BEHAVIOUR
`base_model` should accept, consistently with `add_source(url=...)`:
* A HuggingFace Hub model ID (`"org/model"`) -> **existing behaviour**
* An absolute local path (`"/path/to/model"`) -> **load from disk**
* A `file://` URI (`"file:///path/to/model"`) -> **load from disk**

The same `_local_source_url()` normalisation already used for sources should be applied to `base_model` before it is placed in the proto.
